tests/kernel/common/src/printk.c

```diff
@@ -97,7 +99,9 @@ void printk_test(void)
 			  "%u %2u %4u %8u\n", 42, 42, 42, 42);
 	count += snprintk(ram_console + count, sizeof(ram_console) - count,
 			  "%u %02u %04u %08u\n", 42, 42, 42, 42);
-
+	count += snprintk(ram_console + count, sizeof(ram_console) - count,
+			  "%-8u%-6d%-4x%-2p%8d\n",
```

Comment:
agree with @lpereira, lets adapt the test, making printk always behave as printf is a none goal, printk is a custom version that needs to be small and usable on the smallest of devices.